crates/cow-amm/src/lib.rs

```diff
@@ -0,0 +1,38 @@
+mod cow_amm;
+pub mod cow_amm_constant_product_factory;
+mod event_updater;
+mod indexer;
+
+use ethcontract::Address;
+pub use {event_updater::EventUpdater, indexer::Indexer};
+
+pub trait CowAmm {
```

Comment:
Do we expect different implementations of this trait? If not, why do we need it?